local-cloudbuild.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+#
+# Run the steps of cloudbuild.yaml locally, emulating Google Container
+# Builder functionality.  Does not actually push images to the
+# Container Registry.  Not all functionality is supported.
+#
+# Based on https://cloud.google.com/container-builder/docs/api/build-steps
+
+# System packages
+import argparse
+import getpass
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+# Third party packages
+import yaml
+
+
+# Types
+class CloudBuildError(Exception):
+    pass
+
+
+def main(argv):
+    """Main entrypoint for cli"""
+    parser = argparse.ArgumentParser(
+        description='Process cloudbuild.yaml locally to build Docker images')
+    parser.add_argument(
+        'cloudbuild', type=str, help='Path to cloudbuild.yaml input file')
+    parser.add_argument(
+        '--keep_workspace',
+        type=bool,
+        default=False,
+        help='Retain workspace directory after building')
+    args = parser.parse_args(argv[1:])
```

Comment:
First off - that's is awesome.
Secondly - we (or others) may want to use a specific directory in our CI system for the workspace. (For example, I have a script that downloads the app engine sdk into /tmp by default, but I can also set the path if I wanna cache it)